crates/host/src/oci.rs

```diff
@@ -37,29 +36,47 @@ pub struct Config {
     pub oci_password: Option<String>,
 }
 
+impl From<crate::RegistryAuth> for RegistryAuth {
+    fn from(auth: crate::RegistryAuth) -> Self {
+        match auth {
+            crate::RegistryAuth::Basic(username, password) => Self::Basic(username, password),
+            _ => Self::Anonymous,
+        }
+    }
+}
+
+impl From<&crate::RegistryAuth> for RegistryAuth {
+    fn from(auth: &crate::RegistryAuth) -> Self {
+        match auth {
+            crate::RegistryAuth::Basic(username, password) => {
+                Self::Basic(username.clone(), password.clone())
+            }
+            _ => Self::Anonymous,
+        }
+    }
+}
+
 /// Fetch an OCI path
 #[allow(clippy::missing_errors_doc)] // TODO: document errors
 pub async fn fetch_oci_path(
-    img: &str,
-    registry_settings: &RegistrySettings,
+    img: impl AsRef<str>,
+    auth: &RegistryAuth,
+    allow_latest: bool,
+    allow_insecure: bool,
```

Comment:
These refactors from `RegistrySettings` -> using auth and the two boolean flags feels like a fine refactor for now but might just get reverted if we add one or two more fields. No strongly held opinion here.

Reply:
Yeah, the `OciConfig` should be propagated here, but it's lost along the way.
There's still a bunch of work to do here

Reply:
Updated, PTAL, I think you'll like the new approach